src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Interactivity API: WP_Interactivity_API_Directives_Processor class.
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ */
+
+/**
+ * Class used to iterate over the tags of an HTML string and help process the
+ * directive attributes.
+ *
+ * @access private
+ */
+class WP_Interactivity_API_Directives_Processor extends WP_HTML_Tag_Processor {
+	/**
+	 * Returns the content between two balanced tags.
+	 *
+	 * @access private
```

Comment:
There are multiple `public` methods in this class, but they are all marked as `@access private` — is that intentional? Or should they truly be `private`?

At second glance, seems to be a pattern in the HTML API in general to just have public methods everywhere but mark them as private 😕

Reply:
Yes, it's intentional. They are public because the `WP_Interactivity_API` class needs to access them, but they are marked as private because they are meant to be used by the `WP_Interactivity_API` class only.

Is there another way to do so?

Reply:
I see. Well I suppose that's fine then 👍

Reply:
Ok, thanks Pascal!